### ISSUE TITLE
cni iam policy not attaching to eks managed node group iam role 

### ISSUE THREAD

**Issue by tl-alex-nicot** [terraform-aws-modules/terraform-aws-eks#2966]
module version: "20.8.1"

I belive there is a bug in the logic of attaching the cni policy to the iam role where the conditions do not evaluate correctly. 

in /modules/eks-managed-node-group/main.tf

you have 

```
AmazonEKS_CNI_Policy               = var.iam_role_attach_cni_policy && var.cluster_ip_family == "ipv4" ? local.cni_policy : ""
```

in my managed node group below I have defined iam_role_attach_cni_policy as true and cluster_ip_family as ipv4 but terraform does not want to attach the policy. 
```
eks_managed_node_groups = {
    karpenter = var.karpenter_create_managed_node_group ? {
      name                       = "karpenter-controller"
      use_name_prefix            = true
      instance_types             = var.karpenter_managed_node_group_instance_type
      subnet_ids                 = var.worker_subnets
      ami_id                     = var.eks_ami
      ami_type                   = "AL2_ARM_64"
      platform                   = "linux"
      cluster_ip_family          = "ipv4"
      enable_bootstrap_user_data = true
      bootstrap_extra_args       = "--dns-cluster-ip 169.254.20.10" #NodeLocal ip
      pre_bootstrap_user_data    = <<-EOT
      #!/bin/bash
      modprobe ip_vs
      EOT
      create_iam_role            = true
      iam_role_use_name_prefix   = false
      iam_role_name              = "${var.cluster_name}-karpenter-mng"
      iam_role_attach_cni_policy = true
      min_size                   = var.karpenter_managed_node_group_count
      max_size                   = var.karpenter_managed_node_group_count
      desired_size               = var.karpenter_managed_node_group_count
      key_name                   = "Rescue"
      taints = [{
        "key" : "CriticalAddonsOnly",
        "value" : "Exists",
        "effect" : "NO_EXECUTE"
      }]
      launch_template_tags = merge(var.tags, { "Name" = "${var.cluster_name}-karpenter-controller" })
      tags                 = var.tags

      block_device_mappings = {
        xvda = {
          device_name = "/dev/xvda"
          ebs = {
            volume_size           = var.karpenter_managed_node_group_volume_size
            volume_type           = "gp3"
            encrypted             = true
            kms_key_id            = aws_kms_key.eks-nodes-ebs.arn
            delete_on_termination = true
          }
        }
      }
    }
```

if its something I have done sorry to waste your time.

**Comment by bryantbiggs** [comment]
no this is a bug, fix is coming as we speak

**Comment by tl-alex-nicot** [comment]
forgive me I see your pr if i am already providing  cluster_ip_family  = "ipv4" is it not actually the condition thats wrong 

like it should be 

```
AmazonEKS_CNI_Policy               = var.iam_role_attach_cni_policy == true && var.cluster_ip_family == "ipv4" ? local.cni_policy : ""
```

**Comment by antonbabenko** [comment]
This issue has been resolved in version 20.8.2 :tada:
